theano/tensor/basic.py

```diff
@@ -4585,8 +4591,57 @@ def c_code(self, node, name, inputs, outputs, sub):
         """ % locals()
 
 
+def is_flat(node, outdim=1):
+    """
+    verifies the dimensionality of variable is correct.
+
+    :param node: the theano node on which the dimensionality is checked.
+    :type node: theano.tensor.var.TensorVariable
+
+    :param outdim: the expected dimensionality of node.
+    :type outdim: int
+
+    :returns: the comparison result of node's dim
+              and the expected outdim.
+    """
+    return node.ndim == outdim
+
+
 def flatten(x, outdim=1):
-    return Flatten(outdim)(x)
+    """
+    Reshapes the variable x by keeping
+    the first outdim-1 dimension(s) of x the same,
+    and making the last dimension of x equal to
+    the multiplication of its remaining dimensions.
+
+    :param x: the theano variable that should be reshaped.
+    :type x: theano.tensor.var.TensorVariable
+
+    :param outdim: the number of dimensions of the returned variable
+    :type outdim: int
+
+    :returns: the flattend variable with dimensionality of outdim
+    """
+    outdim = int(outdim)
```

Comment:
I added it to make sure that the outdim is a valid value, but if you think it's unnecessary I can remove it.